javaparser-visitor-generator/src/main/java/com/github/javaparser/generator/visitor/HashCodeVisitorGenerator.java

```diff
@@ -0,0 +1,62 @@
+package com.github.javaparser.generator.visitor;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.stmt.BlockStmt;
+import com.github.javaparser.ast.stmt.Statement;
+import com.github.javaparser.generator.utils.SourceRoot;
+import com.github.javaparser.metamodel.BaseNodeMetaModel;
+import com.github.javaparser.metamodel.JavaParserMetaModel;
+import com.github.javaparser.metamodel.PropertyMetaModel;
+
+import java.util.List;
+
+import static com.github.javaparser.JavaParser.parseStatement;
+import static com.github.javaparser.generator.utils.GeneratorUtils.f;
+
+/**
+ * Generates JavaParser's HashCodeVisitor.
+ */
+public class HashCodeVisitorGenerator extends VisitorGenerator {
+    public HashCodeVisitorGenerator(JavaParser javaParser, SourceRoot sourceRoot, JavaParserMetaModel javaParserMetaModel) {
+        super(javaParser, sourceRoot, "com.github.javaparser.ast.visitor", "HashCodeVisitor", "Integer", "Void", true, javaParserMetaModel);
+    }
+
+    @Override
+    protected void generateVisitMethodBody(BaseNodeMetaModel node, MethodDeclaration visitMethod, List<PropertyMetaModel> propertyMetaModels, CompilationUnit compilationUnit) {
```

Comment:
Disagree, since I don't think exploding a simple algorithm into pieces makes it clearer. It currently shows the decision tree and the different patterns quite clearly, I think.